docs/source/advancedcommands.rst

```diff
@@ -233,3 +233,161 @@ Example Usage
+
+             #construct the model
+             Model.__init__(self, objective, constraints, **kwargs)
+
+.. code-block:: python
+
+     #create model #2
+     from gpkit import Variable, Model
+
+
+     class M2(Model):
+        def __init__(self, **kwargs):
+
+   	    #Make the necessary Variables
+     	    y = Variable(‘y’)
+     	    z = Variable(‘z’)
+
+           #make the constraints
+           constraints = [
+               z >= 2,
+               y*z >= 1,
+           ]
+
+           #declare the objective
+           objective = z*y
+
+          #construct the model
+          Model.__init__(self, objective, constraints, **kwargs)
+
+
+It might be desirable to link ``m1`` and ``m2`` and then solve for the objective ``x*z*y**2``, subject to both the two constraints in ``m1`` as well as the two constraints in ``m2``. Noting ``y`` has the same varkey in both ``m1`` and ``m2``, a ``LinkedConstraintSet`` will automatically link ``y`` between the two models. This This is done below.
+
+.. code-block:: python
+
+     #creating a linked model
+     from gpkit import Variable, Model, LinkedConstraintSet
+     import M1, M2
+
+     #Create the two models
+     m1 = M1()
+     m2 = M2()
+
+     #create a list of submodes
+     self.submodels = [m1, m2]
+
+     #generate a constraint set for the full model
+     constraints = ConstraintSet([self.submodels])
+
+     #create a linked constraint set
+     #now there is only a single y variable
+     lc = LinkedConstraintSet(constraints)
+
+     #create a model, note the cost is x*y*z*y
+     mFull = Model(m1.cost*m2.cost, lc)
+
+     #solve the model
+     mFull.solve()
+
+The cost of the full model is 0.5. If the solution table were to be printed there would only be a single ``y`` variable.
+
+
+Sub In-Place
+--------------
+
+It is also possible to link variables that have a different varkey. Consider a revised ``m2`` presented below. Note that in this version of ``m2``, the variable ``y`` has the varkey ``y2``.
+
+.. code-block:: python
+
+     #create model #2
+     from gpkit import Variable, Model
+
+
+     class M2(Model):
+        def __init__(self, **kwargs):
+
+   	    #Make the necessary Variables
+     	    y = Variable(‘y2’)
+     	    z = Variable(‘z’)
+
+           #make the constraints
+           constraints = [
+               z >= 2,
+               y*z >= 1,
+           ]
+
+           #declare the objective
+           objective = z*y
+
+          #construct the model
+          Model.__init__(self, objective, constraints, **kwargs)
+
+If it was attempted to link ``m1``, from above, and the revised ``m2`` using a ``LinkedConstraintSet``, the two ``y`` variables would not link. This is because they have different varkeys, and a ``LinkedConstraintSet`` only links variables with the same varkey. However, ``subinplace`` can be used to change the varkey of the variable ``y`` in ``m2`` to facilitate linking with a ``LinkedConstraintSet``. This is demonstrated below.
```

Comment:
This looks great, though I'm not a huge fan of string substitutions (since they just rename varkeys, they don't look up the original varkey) and would prefer it if the substitution used `{m1["y"]: m2["y"]}` or whatever the appropriate equivalent is instead.